pyvrp/cpp/ProblemData.h

```diff
@@ -96,6 +108,11 @@ ProblemData::Client const &ProblemData::client(size_t client) const
     return clients_[client];
 }
 
+ProblemData::Route const &ProblemData::route(size_t route) const
+{
+    return routes_[route];
+}
```

Comment:
Like we discussed several times, it's just not always a vehicle:
* Routes may represent walking routes
* In CVRP, we may have a single vehicle delivering all routes one after another
In general, you want to optimize for a set of routes which are not necessarily corresponding to vehicles 1 on 1. I agree we should fix legacy naming inconsistencies.

Nontheless, I agree with the naming confusion and my suggestion would be to call it `RouteData` and then also rename `Client` to `ClientData` (as client is also used to refer to the index at places). Agree?

Reply:
>it's just not always a vehicle

Then why does `ProblemData::Route` currently have a `vehicleCapacity` member - emphasis on _vehicle_? I don't understand the aversion against calling the thing that executes a route a vehicle. A vehicle in this context could be a walking person, a cyclists, a truck, a plane, a spaceship. Making a distinction between:

- The route plan of what's visited in which order (the `Route`);
- The thing executing the route plan (the `Vehicle`);

make a lot of sense to me. Calling the former `Route` and the latter `RouteData` does not alleviate the confusion IMHO.

>In CVRP, we may have a single vehicle delivering all routes one after another
In general, you want to optimize for a set of routes which are not necessarily corresponding to vehicles 1 on 1.

It seems to me that this is something one could solve by pre- and post-processing an instance and its solution.

Reply:
> Then why does `ProblemData::Route` currently have a `vehicleCapacity` member - emphasis on _vehicle_?

That's a good point and I would suggest to call this `capacity`/`loadCapacity`as well (do you want me to change it)?

I can see your points, but I think Route is just more general as the correspondence to Vehicle is not always 1 on 1, sure you can 'pre-/post-process' but that is not the point. We agreed on calling it Route last week (we discussed namespaces as well), now if you want to change it again feel free.

Reply:
>We agreed on calling it Route last week (we discussed namespaces as well), now if you want to change it again feel free.

We discussed what to call the (wrapper) object inside Individual that stores route plans and misc. information; we did not discuss the static data object at all. I recall discussions about things like vehicle and depot indices that need to be stored on that object too, to maintain a link to the static data (which I assumed would still have a list of vehicles). The namespace part was relevant in case that object needed to live outside the Individual class.

The current implementation does not have that wrapper object at all.